### PR TITLE
`GAR`: Use `login-to-gar` in `push-to-gar-docker`

### PR DESCRIPTION
**Pull request by dsotirakis**
Removes duplication - uses one composite action in another.

### CODE REVIEW

File: actions/push-to-gar-docker/action.yaml [hunk 1]
```diff
@@ -89,37 +95,17 @@ runs:
...
         fi
         echo "project=${PROJECT}" >> ${GITHUB_OUTPUT}
-    - name: Get repository name
-      id: get-repository-name
-      shell: bash
-      run: |
-        REPO_NAME=$(echo ${{ github.repository }} | awk -F'/' '{print $2}')
-        echo "repo_name=${REPO_NAME}" >> ${GITHUB_OUTPUT}
+    - name: Login to GAR
+      uses: grafana/shared-workflows/actions/login-to-gar@main
```
Comment (dsotirakis) [inline actions/push-to-gar-docker/action.yaml:99]: will change this from main to pinned in a followup PR which will resolve https://github.com/grafana/shared-workflows/issues/96

Comment (zerok) [reply actions/push-to-gar-docker/action.yaml:99]: But when is that pinning then updated? We don't yet have any real versions here.

Comment (dsotirakis) [reply actions/push-to-gar-docker/action.yaml:99]: yes I was just talking about this to Iain - I guess we should start versioning this thing soon so we can use dependabot for that.

Comment (zerok) [reply actions/push-to-gar-docker/action.yaml:99]: Agreed 🙂 Until we have that, though, we should refrain from internal pinning IMO.

**Review by zerok — APPROVED** [review]

File: actions/push-to-gar-docker/action.yaml (hunk 1 above)
Comment (iainlane) [reply actions/push-to-gar-docker/action.yaml:99]: https://github.com/grafana/shared-workflows/issues/68

**Pull request merged**